### PR TITLE
Answer a question on its own page

### PR DESCRIPTION
**Pull request by rayanb**
I have implemented the answer form, which is a partial with locals now, on the question, we don't need a new route anymore !

### CODE REVIEW

File: app/controllers/answers_controller.rb [hunk 1]
```diff
@@ -1,11 +1,33 @@
 class AnswersController < ApplicationController
 
-  def index
+  def create
+    question = Question.find(params[:question_id])
+    answer = question.answers.new(content: params[:answer][:content], user_id: session[:user_id])
+    if answer.save
+      # Ajax
+      redirect_to question
+    else
```
Comment (sgharms) [inline app/controllers/answers_controller.rb:9]: if you're doing the same thing on either side of the if/else, why have the if/else at all?

File: spec/controllers/answers_controller_spec.rb [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
-    it "GET questions/x/answers"
-    it "GET questions/x/answers/y"
-  end
+  let(:user){User.create(username: "Leo Dongchul", password: '123', email: "therio@gmail.com")}
+  let(:question) {Question.create(title: "Who is DongChul ?", content: "He is mysterious and mindful, who is he, really ?", category_id: 1, user: user)}
+
+  context 'hey' do
+    before(:each) do
+      user
+      question
```
Comment (sgharms) [inline spec/controllers/answers_controller_spec.rb:11]: NO NO NO NO NO

File: app/controllers/answers_controller.rb (hunk 1 above)
Comment (rayanb) [inline app/controllers/answers_controller.rb:9]: it is temporary, I'll eventually add an error message with ajax if it goes wrong

File: spec/controllers/answers_controller_spec.rb (hunk 2 above)
Comment (sgharms) [inline spec/controllers/answers_controller_spec.rb:11]: You should not have to do this.  If you _do_, there are better ways to force early creation.  See `let!` versus `let` in RSpec.

**Pull request merged**